tests/EDD/test_fit.py

```diff
@@ -26,44 +29,56 @@ def generate_config(tmp_path: Path, test_data_path: Union[Path, str]) -> dict:
         "nbPeaksInBoxes": nb_peaks_in_boxes,
         "rangeFitHD": fit_ranges_h,
         "rangeFitVD": fit_ranges_v,
+        "detectorSliceIndex": None if scan_length is None else scan_length // 2,
     }
 
 
 def generate_input_files(
-    tmp_path: Path,
-    test_data_path: Union[Path, str],
+    tmp_path: Path, test_data_path: Union[Path, str], scan_length: Optional[int] = None
 ) -> dict:
-    cfg = generate_config(tmp_path, test_data_path)
+    cfg = generate_config(tmp_path, test_data_path, scan_length)
     sample, dataset = cfg["sample"], cfg["dataset"]
     n_scan = cfg["scanNumber"]
     name_h, name_v = (
         cfg["nameHorizontalDetector"],
         cfg["nameVerticalDetector"],
     )
+
+    detector_dim = scan_length if scan_length is not None else 1
```

Comment:
You really love the ternary conditional operator.  I personally find them impossible to read.

Reply:
Well yeah, I find them easier to read than a branching if. I guess it is a matter a taste.